### PR TITLE
chore: migrate navigations to Progress

### PR DESCRIPTION
**Pull request by dgozman**
This also adds navigation-related logging and tests.

### CODE REVIEW

**Review by yury-s — APPROVED** [review]

File: src/progress.ts [hunk 1]
```diff
@@ -41,13 +44,15 @@ export class Progress {
...
       const result = await Promise.race([promise, cancelPromise]);
+      cancelPromise.catch(e => null);
       clearTimeout(timer);
       progress._running = false;
       progress._logRecording = [];
       return result;
     } catch (e) {
       resolveCancelation();
       rewriteErrorMessage(e, e.message + formatLogRecording(progress._logRecording, apiName));
+      cancelPromise.catch(e => null);
```
[the remark this answers is not in the document]
Comment (dgozman) [reply src/progress.ts:55]: Done.

File: src/frames.ts [hunk 2]
```diff
@@ -350,55 +341,64 @@ export class Frame {
...
-      await frameTask.waitForSpecificDocument(navigateResult.newDocumentId);
-    } else {
-      await sameDocumentPromise;
-    }
-    const request = (navigateResult && navigateResult.newDocumentId) ? frameTask.request(navigateResult.newDocumentId) : null;
-    await frameTask.waitForLifecycle(options.waitUntil === undefined ? 'load' : options.waitUntil);
-    frameTask.done();
-    return request ? request._finalRequest().response() : null;
+    return Progress.runCancelableTask(async progress => {
+      progress.log(dom.inputLog, `${progress.apiName}("${url}"), waiting until "${options.waitUntil || 'load'}"`);
```
Comment (pavelfeldman) [inline src/frames.ts:345]: Is inputLog now apiLog?

File: src/frames.ts [hunk 3]
```diff
@@ -350,55 +341,64 @@ export class Frame {
...
-      sameDocumentPromise.catch(e => {});
-      await frameTask.waitForSpecificDocument(navigateResult.newDocumentId);
-    } else {
-      await sameDocumentPromise;
-    }
-    const request = (navigateResult && navigateResult.newDocumentId) ? frameTask.request(navigateResult.newDocumentId) : null;
-    await frameTask.waitForLifecycle(options.waitUntil === undefined ? 'load' : options.waitUntil);
-    frameTask.done();
-    return request ? request._finalRequest().response() : null;
+    return Progress.runCancelableTask(async progress => {
```
Comment (pavelfeldman) [inline src/frames.ts:344]: `this._page._runCancelableTask(() => {}, options)`

File: src/frames.ts [hunk 4]
```diff
@@ -350,55 +341,64 @@ export class Frame {
...
   }
 
   async waitForLoadState(state: types.LifecycleEvent = 'load', options: types.TimeoutOptions = {}): Promise<void> {
-    const frameTask = new FrameTask(this, options);
+    return Progress.runCancelableTask(progress => this._waitForLoadState(progress, state), options, this._page, this._page._timeoutSettings.navigationTimeout());
+  }
+
+  async _waitForLoadState(progress: Progress, state: types.LifecycleEvent): Promise<void> {
+    const frameTask = new FrameTask(this, progress);
+    frameTask.cancelProgressOnFailure();
```
Comment (pavelfeldman) [inline src/frames.ts:401]: What does this mean?

File: src/frames.ts [hunk 5]
```diff
@@ -1008,38 +1020,24 @@ export class FrameTask {
...
-    throw error;
-  }
-
   request(documentId: string): network.Request | undefined {
     return this._requestMap.get(documentId);
   }
 
   waitForSameDocumentNavigation(url?: types.URLMatch): Promise<void> {
-    return this.raceAgainstFailures(new Promise((resolve, reject) => {
+    return new Promise((resolve, reject) => {
```
Comment (pavelfeldman) [inline src/frames.ts:1028]: This now looks weird

File: src/frames.ts (hunk 3 above)
Comment (dgozman) [reply src/frames.ts]: I will follow up.

File: src/frames.ts (hunk 5 above)
Comment (dgozman) [reply src/frames.ts:1030]: I will try to follow up :)

**Pull request merged**